test/integration/agent_hosting_hook_deploy_test.go

```diff
@@ -322,6 +325,56 @@ var _ = ginkgo.Describe("Agent hook deploy", func() {
 		_, err = hubWorkClient.WorkV1().ManifestWorks(hostingClusterName).
 			UpdateStatus(context.Background(), work, metav1.UpdateOptions{})
 		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+		// addon only has pending finalizer, and status is updated
+		gomega.Eventually(func() error {
+			addon, err := hubAddonClient.AddonV1alpha1().ManagedClusterAddOns(managedClusterName).
+				Get(context.Background(), testHostedAddonImpl.name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if len(addon.Finalizers) != 1 {
+				return fmt.Errorf("addon is expected to only 1 finalizer,but got %v", len(addon.Finalizers))
+			}
+			if addon.Finalizers[0] != "pending" {
+				return fmt.Errorf("addon is expected to only pending finalizer,but got %v", len(addon.Finalizers))
+			}
+			if !meta.IsStatusConditionTrue(addon.Status.Conditions, addonapiv1alpha1.ManagedClusterAddOnHookManifestCompleted) {
+				return fmt.Errorf("addon HookManifestCompleted condition is expecte to true, but got false")
+			}
+			return nil
+		}, eventuallyTimeout, eventuallyInterval).ShouldNot(gomega.HaveOccurred())
+
+		// update addon to trigger reconcile 3 times, and per-delete manifestwork should be deleted and not be re-created
+		for i := 0; i < 3; i++ {
+			gomega.Eventually(func() error {
+				addon, err := hubAddonClient.AddonV1alpha1().ManagedClusterAddOns(managedClusterName).
+					Get(context.Background(), testHostedAddonImpl.name, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				addon.Labels = map[string]string{"test": fmt.Sprintf("%d", i)}
+				_, err = hubAddonClient.AddonV1alpha1().ManagedClusterAddOns(managedClusterName).
+					Update(context.Background(), addon, metav1.UpdateOptions{})
```

Comment:
consider putting the getting/updating into a retry block since it may fail with a conflict.

Reply:
done